src/main/java/org/mineacademy/fo/model/HookManager.java

```diff
@@ -1662,6 +1691,18 @@ public static void sendDiscordMessage(final String channel, @NonNull final Strin
 //
 // ------------------------------------------------------------------------------------------------------------
 
+class AdvancedVanishHook {
+
+	boolean isVanished(final Player player) {
+		return player != null && AdvancedVanishAPI.INSTANCE.isPlayerVanished(player);
+	}
+
+	void setVanished(Player player, boolean vanished) {
+		if (player != null && isVanished(player) != vanished)
+			setVanished(player, false);

```

Comment:
this method calls itself and causes infinite loop, next time please test

Reply:
> this method calls itself and causes infinite loop, next time please test

This looks weird, I did run tests and didn't experience any issues